### PR TITLE
Matthew Hardman

### PR DESCRIPTION
**Pull request by mattshardman**

### CODE REVIEW

File: starwars/src/App.js [hunk 1]
```diff
@@ -1,38 +1,84 @@
...
     super();
     this.state = {
-      starwarsChars: []
+      nextPage: null,
+      starWarsChars: [],
     };
   }
 
   componentDidMount() {
+    document.addEventListener('scroll', () => {
```
Comment (Velsu) [inline starwars/src/App.js:28]: I wonder if this listener should be removed after `componentWillUnmount` is called with `document.removeEventListener`.

**Review by Velsu — APPROVED** [review]
### React Sprint Review

Matt as usual You are deliving exceptional project for sprint challenge or any other assingment. You successfully completed the MVP by fetching data from 3rd party API and displayed it on the webpage. You have properly split functionality into smaller components and passed necessary data as props.

Your project certainly has unique style and unique way of loading new dataset by scrolling down as a stretch goal. Website is somewhat responsive but i believe it could look better with more effort.

Your code is clean and Your git workflow is fluid.

#### Overall:
- Able to fetch and display data (MVP)
- Possible to load another set of data by scrolling down (stretch)
- Clean and modular code with multiple components
- Implemented styling
- Fluid git workflow

#### Possible improvements:
- Responsiveness could be a little better on mobile view
- It would be great if movies displayed as titles and not as links
- Missing answers to the questions

### Summary

Matt this is an exceptional job and You have displayed mastery on the subject 👍

**Comment by mattshardman** [comment]
Not sure why my answers hadn't pushed?

- [ ] What is React JS and what problems does it try and solve? Support your answer with concepts introduced in class and from your personal research on the web.

* React is a JS library for building user interfaces. It solves a number of problems, namely:
  * - By having a "virtual dom" react limits DOM manipulation by only updating what needs to be updated
  * - React is modular and reusable - by breaking sites down into small components React makes it easy to reuse components and scale
  * - React makes it easy to create single page applications, meaning they can increase performance as the browser does not have to call a server each time a page is loaded. (This comes with other issues such as initial render speed - but their are methods around this such as server side rendering etc...)

- [ ] What does it mean to _think_ in react?

* To think in react means to think of websites as small reusable components. Applications should be broken down into a component hierarchy, using "dumb" functional components to display data, and class components to control the state of an application.

- [ ] Describe state.

* State is an object that describes how a component renders and behaves. State stores dynamic data, the fact that is dynamic is important as this means it can be updated and allows components to be interactive. State is only used in Class components (with the exception of React hooks introduced in React 16.8);

- [ ] Describe props.

* Props are data passed to a component from another component. Unlike state, the component does not alter the props it simply displays them (or uses them to determine how the component should be rendered).

**Comment by Velsu** [comment]
Thanks for the update Matt 👍

**Pull request merged**